sql/catalog/lease: TestDescriptorRefreshOnRetry failed

[(sql/catalog/lease).TestDescriptorRefreshOnRetry failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2001700&tab=buildLog) on [master@c5e5f5f88a6b5dfb89585e53a0a39414b225717c](https://github.com/cockroachdb/cockroach/commits/c5e5f5f88a6b5dfb89585e53a0a39414b225717c):

```
I200610 04:31:27.818986 21922 gossip/gossip.go:406  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:34947" > attrs:<> locality:<> ServerVersion:<major_val:20 minor_val:1 patch:0 unstable:7 > build_tag:"v20.2.0-alpha.00000000-2193-gc5e5f5f88a" started_at:1591763487818977363 cluster_name:"" sql_address:<network_field:"tcp" address_field:"127.0.0.1:43249" > 
W200610 04:31:27.825000 22070 kv/kvserver/store.go:1651  [n1,s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I200610 04:31:27.826177 21995 server/server.go:1330  [n1] node connected via gossip
I200610 04:31:27.827499 21922 server/node.go:412  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=18 KiB), ranges=34, leases=2, queries=0.00, writes=0.00, bytesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=166.00 pMax=15517.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00}
I200610 04:31:27.827684 21922 kv/kvserver/stores.go:236  [n1] read 0 node addresses from persistent storage
I200610 04:31:27.827791 21922 server/node.go:468  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I200610 04:31:27.827915 21922 server/server.go:1426  [n1] starting https server at 127.0.0.1:46201 (use: 127.0.0.1:46201)
I200610 04:31:27.838905 21922 server/server.go:1431  [n1] starting postgres server at 127.0.0.1:43249 (use: 127.0.0.1:43249)
I200610 04:31:27.839080 21922 server/server.go:1433  [n1] starting grpc server at 127.0.0.1:34947
I200610 04:31:27.839178 21922 server/server.go:1434  [n1] advertising CockroachDB node at 127.0.0.1:34947
I200610 04:31:27.854375 22112 sql/temporary_schema.go:473  [n1] running temporary object cleanup background job
I200610 04:31:27.856750 22112 sql/temporary_schema.go:508  [n1] found 0 temporary schemas
I200610 04:31:27.856794 22112 sql/temporary_schema.go:511  [n1] early exiting temporary schema cleaner as no temporary schemas were found
I200610 04:31:27.856813 22112 sql/temporary_schema.go:512  [n1] completed temporary object cleanup job
I200610 04:31:27.856827 22112 sql/temporary_schema.go:595  [n1] temporary object cleaner next scheduled to run at 2020-06-10 05:01:27.85374925 +0000 UTC
I200610 04:31:27.857127 22136 kv/kvserver/replica_rangefeed.go:608  [n1,s1,r6/1:/Table/{SystemCon…-11}] RangeFeed closed timestamp is empty
I200610 04:31:27.862087 22111 sql/event_log.go:132  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:127.0.0.1:34947 Attrs: Locality: ServerVersion:20.1-7 BuildTag:v20.2.0-alpha.00000000-2193-gc5e5f5f88a StartedAt:1591763487818977363 LocalityAddress:[] ClusterName: SQLAddress:127.0.0.1:43249} ClusterID:f5820f32-58cb-4f25-8eff-5573c1c722a0 StartedAt:1591763487818977363 LastUp:1591763487818977363}
I200610 04:31:27.864295 22155 sql/event_log.go:132  [n1,intExec=optInToDiagnosticsStatReporting] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:root}
I200610 04:31:27.876687 22161 sql/event_log.go:132  [n1,intExec=set-setting] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:20.1-7 User:root}
I200610 04:31:27.906712 22579 sql/event_log.go:132  [n1,intExec=initializeClusterSecret] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:c5508035-5871-4099-a244-385e7e60bd63 User:root}
I200610 04:31:27.910572 22110 server/server_update.go:55  [n1] no need to upgrade, cluster already at the newest version
I200610 04:31:27.912909 22586 sql/event_log.go:132  [n1,intExec=create-default-db] Event: "create_database", target: 50, info: {DatabaseName:defaultdb Statement:CREATE DATABASE IF NOT EXISTS defaultdb User:root}
I200610 04:31:27.920445 22672 sql/event_log.go:132  [n1,intExec=create-default-db] Event: "create_database", target: 51, info: {DatabaseName:postgres Statement:CREATE DATABASE IF NOT EXISTS postgres User:root}
I200610 04:31:27.937831 21922 server/server_sql.go:645  [n1] done ensuring all necessary migrations have run
I200610 04:31:27.938354 21922 server/server.go:1747  [n1] serving sql connections
I200610 04:31:27.942271 22853 sqlmigrations/migrations.go:727  [n1] starting wait for upgrade finalization before schema change job migration
I200610 04:31:27.942474 22853 sqlmigrations/migrations.go:759  [n1] starting schema change job migration
I200610 04:31:27.949713 22853 sqlmigrations/migrations.go:1177  [schema-change-job-migration] evaluating tables for creating jobs
I200610 04:31:27.950174 22853 sqlmigrations/migrations.go:779  [n1] schema change job migration completed
I200610 04:31:27.970888 22757 sql/event_log.go:132  [n1,client=127.0.0.1:49080,hostssl,user=root] Event: "create_database", target: 52, info: {DatabaseName:t Statement:CREATE DATABASE t User:root}
I200610 04:31:27.982217 22757 sql/event_log.go:132  [n1,client=127.0.0.1:49080,hostssl,user=root] Event: "create_table", target: 53, info: {TableName:t.public.foo Statement:CREATE TABLE t.public.foo (v INT8) User:root}
I200610 04:31:27.985909 23010 kv/kvserver/replica_command.go:413  [n1,split,s1,r34/1:/{Table/38-Max}] initiating a split of this range at key /Table/53 [r35] (zone config)
I200610 04:31:28.022589 21922 util/stop/stopper.go:539  quiescing
W200610 04:31:28.023006 22928 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
I200610 04:31:28.025397 22038 kv/kvserver/queue.go:580  [n1,s1,r9/1:/Table/1{3-4}] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200610 04:31:28.025624 22038 kv/kvserver/queue.go:580  [n1,s1,r9/1:/Table/1{3-4}] rate limited in MaybeAdd (merge): node unavailable; try another peer
W200610 04:31:28.025734 22038 kv/kvserver/store.go:1840  [n1,s1,r9/1:/Table/1{3-4}] unable to gossip on capacity change: node unavailable; try another peer
W200610 04:31:28.026221 22935 kv/txn.go:607  [intExec=lease-release] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
W200610 04:31:28.026314 22934 sql/catalog/lease/lease.go:291  error releasing lease &{'5' '\x01' {{%!q(uint64=873617000) %!q(int64=63727360576) %!q(*time.Location=<nil>)}}}: lease-release: result is ambiguous (server shutdown)
W200610 04:31:28.026739 23010 kv/txn.go:607  [n1,split,s1,r34/1:/{Table/38-Max}] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
E200610 04:31:28.026841 23010 kv/kvserver/queue.go:1082  [n1,split,s1,r34/1:/{Table/38-Max}] unable to split [n1,s1,r34/1:/{Table/38-Max}] at key "/Table/53": split at key /Table/53 failed: node unavailable; try another peer
E200610 04:31:28.026903 22933 kv/kvclient/kvcoord/txn_interceptor_committer.go:449  making txn commit explicit failed for "unnamed" meta={id=30e76887 key=/Table/11/1/53/1/2020-06-10T04:36:19.888026Z/1/0 pri=100.00000000 epo=0 ts=1591763488.000552502,0 min=1591763488.000552502,0 seq=2} lock=true stat=STAGING rts=1591763488.000552502,0 wto=false max=1591763488.000552502,0 ifw=1: result is ambiguous (server shutdown)
W200610 04:31:28.029411 22757 kv/txn.go:607  [n1,client=127.0.0.1:49080,hostssl,user=root] failure aborting transaction: node unavailable; try another peer; abort caused by: connExecutor closing
W200610 04:31:28.029476 22757 sql/catalog/lease/lease.go:1128  error: node unavailable; try another peer, not releasing lease: &{'5' '\x01' {{%!q(uint64=888026000) %!q(int64=63727360579) %!q(*time.Location=<nil>)}}}
--- FAIL: TestDescriptorRefreshOnRetry (0.30s)
    lease_test.go:879: invalid descriptor acquisition counts = 2, 0


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestDescriptorRefreshOnRetry PKG=./pkg/sql/catalog/lease TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestDescriptorRefreshOnRetry.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>


Jira issue: CRDB-4161

[(sql/catalog/lease).TestDescriptorRefreshOnRetry failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2070728&tab=buildLog) on [master@cff2542785117d9c85e9c0b3daba4371f0a05795](https://github.com/cockroachdb/cockroach/commits/cff2542785117d9c85e9c0b3daba4371f0a05795):

```
W200707 17:01:23.874868 21089 server/status/runtime.go:310  [n?] Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I200707 17:01:23.889465 21089 server/server.go:803  [n?] monitoring forward clock jumps based on server.clock.forward_jump_check_enabled
I200707 17:01:23.901782 21089 server/config.go:600  [n?] 1 storage engine initialized
I200707 17:01:23.902009 21089 server/config.go:603  [n?] Pebble cache size: 128 MiB
I200707 17:01:23.902309 21089 server/config.go:603  [n?] store 0: in-memory, size 0 B
I200707 17:01:23.911810 21089 server/server.go:1301  [n?] **** add additional nodes by specifying --join=127.0.0.1:39247
I200707 17:01:23.911922 21089 server/init.go:173  [n?] no stores bootstrapped and --join flag specified, awaiting init command or join with an already initialized node.
I200707 17:01:23.911988 21089 server/init.go:193  [n?] **** cluster 1447211b-a622-4891-8323-1998d5681613 has been created
I200707 17:01:23.912128 20431 server/server.go:1383  [n1] connecting to gossip network to verify cluster ID "1447211b-a622-4891-8323-1998d5681613"
I200707 17:01:23.912106 21089 gossip/gossip.go:403  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:39247" > attrs:<> locality:<> ServerVersion:<major_val:20 minor_val:1 patch:0 unstable:9 > build_tag:"v20.2.0-alpha.1-967-gcff2542785" started_at:1594141283912098749 cluster_name:"" sql_address:<network_field:"tcp" address_field:"127.0.0.1:36017" > 
I200707 17:01:23.920016 21089 server/node.go:412  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=18 KiB), ranges=34, leases=0, queries=0.00, writes=0.00, bytesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=166.00 pMax=15517.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00 pMax=0.00}
I200707 17:01:23.920214 21089 kv/kvserver/stores.go:236  [n1] read 0 node addresses from persistent storage
I200707 17:01:23.920287 21089 server/node.go:468  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I200707 17:01:23.920352 21089 server/server.go:1482  [n1] starting https server at 127.0.0.1:34905 (use: 127.0.0.1:34905)
I200707 17:01:23.920377 21089 server/server.go:1487  [n1] starting postgres server at 127.0.0.1:36017 (use: 127.0.0.1:36017)
I200707 17:01:23.920403 21089 server/server.go:1489  [n1] starting grpc server at 127.0.0.1:39247
I200707 17:01:23.920420 21089 server/server.go:1490  [n1] advertising CockroachDB node at 127.0.0.1:39247
I200707 17:01:23.921593 20431 server/server.go:1386  [n1] node connected via gossip
W200707 17:01:23.922226 21354 kv/kvserver/store.go:1656  [n1,s1,r6/1:/Table/{SystemCon…-11}] could not gossip system config: [NotLeaseHolderError] r6: replica (n1,s1):1 not lease holder; lease holder unknown
I200707 17:01:23.975676 21438 sql/temporary_schema.go:473  [n1] running temporary object cleanup background job
I200707 17:01:23.977067 21484 kv/kvserver/replica_rangefeed.go:608  [n1,s1,r6/1:/Table/{SystemCon…-11}] RangeFeed closed timestamp is empty
I200707 17:01:23.977851 21437 sql/event_log.go:132  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:127.0.0.1:39247 Attrs: Locality: ServerVersion:20.1-9 BuildTag:v20.2.0-alpha.1-967-gcff2542785 StartedAt:1594141283912098749 LocalityAddress:[] ClusterName: SQLAddress:127.0.0.1:36017} ClusterID:1447211b-a622-4891-8323-1998d5681613 StartedAt:1594141283912098749 LastUp:1594141283912098749}
I200707 17:01:23.979104 21438 sql/temporary_schema.go:508  [n1] found 0 temporary schemas
I200707 17:01:23.979141 21438 sql/temporary_schema.go:511  [n1] early exiting temporary schema cleaner as no temporary schemas were found
I200707 17:01:23.979157 21438 sql/temporary_schema.go:512  [n1] completed temporary object cleanup job
I200707 17:01:23.979172 21438 sql/temporary_schema.go:595  [n1] temporary object cleaner next scheduled to run at 2020-07-07 17:31:23.975636258 +0000 UTC
I200707 17:01:23.979892 21488 sql/event_log.go:132  [n1,intExec=optInToDiagnosticsStatReporting] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:root}
I200707 17:01:23.986545 21654 sql/event_log.go:132  [n1,intExec=set-setting] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:20.1-9 User:root}
I200707 17:01:23.999539 21889 sql/event_log.go:132  [n1,intExec=initializeClusterSecret] Event: "set_cluster_setting", target: 0, info: {SettingName:cluster.secret Value:5c055f08-611c-48b1-8933-0d1bd67ad155 User:root}
I200707 17:01:24.016654 21897 sql/event_log.go:132  [n1,intExec=create-default-db] Event: "create_database", target: 50, info: {DatabaseName:defaultdb Statement:CREATE DATABASE IF NOT EXISTS defaultdb User:root}
I200707 17:01:24.020188 21964 sql/event_log.go:132  [n1,intExec=create-default-db] Event: "create_database", target: 51, info: {DatabaseName:postgres Statement:CREATE DATABASE IF NOT EXISTS postgres User:root}
I200707 17:01:24.034736 21436 server/server_update.go:55  [n1] no need to upgrade, cluster already at the newest version
I200707 17:01:24.054592 21089 server/server_sql.go:643  [n1] done ensuring all necessary migrations have run
I200707 17:01:24.054657 21089 server/server.go:1834  [n1] serving sql connections
I200707 17:01:24.056720 22072 sqlmigrations/migrations.go:727  [n1] starting wait for upgrade finalization before schema change job migration
I200707 17:01:24.056987 22072 sqlmigrations/migrations.go:759  [n1] starting schema change job migration
I200707 17:01:24.080723 22072 sqlmigrations/migrations.go:1177  [schema-change-job-migration] evaluating tables for creating jobs
I200707 17:01:24.080769 22072 sqlmigrations/migrations.go:779  [n1] schema change job migration completed
I200707 17:01:24.085385 22078 sql/event_log.go:132  [n1,client=127.0.0.1:60368,hostssl,user=root] Event: "create_database", target: 52, info: {DatabaseName:t Statement:CREATE DATABASE t User:root}
I200707 17:01:24.095479 22078 sql/event_log.go:132  [n1,client=127.0.0.1:60368,hostssl,user=root] Event: "create_table", target: 53, info: {TableName:t.public.foo Statement:CREATE TABLE t.public.foo (v INT8) User:root}
I200707 17:01:24.101281 22211 kv/kvserver/replica_command.go:414  [n1,split,s1,r34/1:/{Table/38-Max}] initiating a split of this range at key /Table/53 [r35] (zone config)
    TestDescriptorRefreshOnRetry: lease_test.go:881: invalid descriptor acquisition counts = 2, 0
I200707 17:01:24.111427 21089 util/stop/stopper.go:539  quiescing
W200707 17:01:24.111650 22216 kv/kvserver/intentresolver/intent_resolver.go:744  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
W200707 17:01:24.114309 22078 kv/txn.go:610  [n1,client=127.0.0.1:60368,hostssl,user=root] failure aborting transaction: node unavailable; try another peer; abort caused by: connExecutor closing
--- FAIL: TestDescriptorRefreshOnRetry (0.24s)


Failed to find issue assignee: 
old friend vivekmenezes is friendless
```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestDescriptorRefreshOnRetry PKG=./pkg/sql/catalog/lease TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestDescriptorRefreshOnRetry.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/catalog/lease/lease_test.go
+++ b/pkg/sql/catalog/lease/lease_test.go
@@ -890,8 +890,6 @@ CREATE TABLE t.foo (v INT);
 func TestDescriptorRefreshOnRetry(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 
-	skip.WithIssue(t, 50037)
-
 	params := createTestServerParams()
 
 	fooAcquiredCount := int32(0)

